backend/schema/schema_test.go

```diff
@@ -481,3 +481,86 @@ var testSchema = MustValidate(&Schema{
 		},
 	},
 })
+
+func TestValidateDependencies(t *testing.T) {
+	// one <--> two, cyclical
+	_, err := ParseString("", `
+		module one {
+			verb one(builtin.Empty) builtin.Empty
+				calls two.two
+		}
+
+		module two {
+			verb two(builtin.Empty) builtin.Empty
+				calls one.one
+		}
+	`)
+	assert.Error(t, err)
+
+	// one --> two --> three, noncyclical
+	_, err = ParseString("", `
+		module one {
+			verb one(builtin.Empty) builtin.Empty
+				calls two.two
+		}
+
+		module two {
+			verb two(builtin.Empty) builtin.Empty
+				calls three.three
+		}
+
+		module three {
+			verb three(builtin.Empty) builtin.Empty
+		}
+	`)
+	assert.NoError(t, err)
+
+	// one --> two --> three -> one, cyclical
+	_, err = ParseString("", `
+		module one {
+			verb one(builtin.Empty) builtin.Empty
+				calls two.two
+		}
+
+		module two {
+			verb two(builtin.Empty) builtin.Empty
+				calls three.three
+		}
+
+		module three {
+			verb three(builtin.Empty) builtin.Empty
+			calls one.one
+		}
+	`)
+	assert.Error(t, err)
+
+	// one --> one, this is allowed
+	_, err = ParseString("", `
+		module one {
+			verb a(builtin.Empty) builtin.Empty
+				calls one.b
+
+			verb b(builtin.Empty) builtin.Empty
+				calls one.a
+		}
+	`)
+	assert.NoError(t, err)
+
+	// one.a --> two.a
+	// one.b <---
+	// cyclical (does not depend on verbs used)
+
+	_, err = ParseString("", `
```

Comment:
This is awesome. I would suggest converting this into a "table driven test", which will remove some duplication. They're quite common in Go. 

It would be something like:

```go
tests := []struct{
  name string
  schema string
  err string
}{
// ...
}

for _, test := range tests {
  t.Run(test.name, func(t *testing.T) {
    // ...
    assert.EqualError(t, err, test.error)
  })
```